### PR TITLE
fix abyparty debugging output

### PR DESCRIPTION
**Pull request by sebastianst**
some errors sneaked in during the `std::`ification. Also, a pp flag was erroneously checked with `#if` instead of `#ifdef`.

### CODE REVIEW

File: src/abycore/aby/abyparty.cpp [hunk 1]
```diff
@@ -453,9 +453,9 @@ BOOL ABYParty::ThreadSendValues() {
 			snd_buf_size_total += sndbytes[j][i];
 			//m_tPartyChan->send(sendbuf[j][i], sndbytes[j][i]);
 #ifdef DEBUGCOMM
-			std::cout_mutex.lock();
```
Comment (lenerd) [inline src/abycore/aby/abyparty.cpp:456]: Ooops, I guess that is my fault using too much `s/../../`.

**Comment by lenerd** [comment]
It is really great that `#define FOO`/`#ifdef FOO` and `#define BAR 1`/`#if BAR` are mixed all over the code.
Maybe we can replace these in the future with something like
```c++
if constexpr (aby_options.foo) {
    ...
}
```
with a const struct `aby_options` known at compile time. The compiler would see all the code even if the option is disabled which would help to prevent such mistakes.

**Comment by dd23** [comment]
Very good idea! 
If these `aby_options` are known at compile time, we achieve the same thing (performance by not running unnecessary code) as with the  #defines and increase code quality.

**Comment by sebastianst** [comment]
I agree, great idea! Less use of the preprocessor. However, we still need a way to specify some options at compile time to have several build flavors, like debugging and production (see `-DDEBUG` etc.).

**Pull request merged**